### PR TITLE
[doc] Add switch task demo

### LINKED ISSUE #8029: [Feature][Document] Add example and notice about task type Switch
Due to our tasks task description is not containing some detail, and I think we should add a new section notice and give an example to the task switch in the document https://dolphinscheduler.apache.org/en-us/docs/latest/user_doc/guide/task/switch.html

### PR DESCRIPTION
**Pull request by huangchenguang123**
close #8029

### CODE REVIEW

File: docs/docs/zh/guide/task/switch.md [hunk 1]
```diff
@@ -13,25 +15,34 @@ Switch是一个条件判断节点，依据[全局变量](../parameter/global.md)
 - 描述信息：描述该节点的功能。
 - 任务优先级：worker线程数不足时，根据优先级从高到低依次执行，优先级一样时根据先进先出原则执行。
```
Comment (QuakeWang) [inline docs/docs/zh/guide/task/switch.md]:
```suggestion
- 任务优先级：worker 线程数不足时，根据优先级从高到低依次执行，优先级一样时根据先进先出原则执行。
```
It is recommended to add spaces between Chinese characters, numbers, and English. The same goes for the content behind it.

**Review by QuakeWang — COMMENTED** [review]
And the screenshots corresponding to the documentation are recommended to be in light mode. Thanks a lot : )

File: docs/docs/zh/guide/task/switch.md [hunk 2]
```diff
@@ -1,10 +1,12 @@
 # Switch
 
-Switch是一个条件判断节点，依据[全局变量](../parameter/global.md)的值和用户所编写的表达式判断结果执行对应分支。
```
Comment (QuakeWang) [inline docs/docs/zh/guide/task/switch.md:3]: It is recommended to use the original expression, which is more complete.

File: docs/docs/zh/guide/task/switch.md [hunk 3]
```diff
@@ -13,25 +15,34 @@ Switch是一个条件判断节点，依据[全局变量](../parameter/global.md)
...
-- 任务A中通过[全局变量](../parameter/global.md)定义了名为`id`的全局变量，声明方式为`${setValue(id=1)}`
-- 任务B增加条件，使用上游声明的全局变量实现条件判断（注意：只要直接、非直接上游在switch运行前对全局变量赋值，switch运行时就可以获取该全局变量）。下面我们想要实现当id为1时，运行任务C，其他运行任务D
-  - 配置当全局变量`id=1`时，运行任务C。则在任务B的条件中编辑`${id} == 1`，分支流转选择`C`
-  - 对于其他任务，在分支流转中选择`D`
+![switch_01](/img/tasks/demo/switch_01.png)
 
-最终switch任务的配置如下
+### 2、设置条件
+配置条件和默认分支，满足条件会走指定分支，都不满足则走默认分支。
+条件使用了全局变量，请参考[全局变量](../parameter/global.md)。
```
Comment (QuakeWang) [inline docs/docs/zh/guide/task/switch.md:46]: Some details can be added here. For example, explain the value of the global variable set in this example, and the branch corresponding to the execution task.

File: docs/docs/zh/guide/task/switch.md [hunk 4]
```diff
@@ -13,25 +15,34 @@ Switch是一个条件判断节点，依据[全局变量](../parameter/global.md)
...
+- 延时执行时间：任务延迟执行的时间。
+- 超时告警：勾选超时告警、超时失败，当任务超过"超时时长"后，会发送告警邮件并且任务执行失败。
+- 条件：可以为switch任务配置多个条件，当条件满足时，就会执行指定的分支，可以配置多个不同的条件来满足不同的业务，使用字符串判断时需要使用""。
+- 分支流转：默认的流转内容，当**条件**中的内容为全部不符合要求时，则运行**分支流转**中指定的分支。
 
 ## 详细流程
+这里使用1个 switch 任务以及3个 shell 任务来演示。
 
-假设我们三个任务，其依赖关系是 `A -> B -> [C, D]` 其中task_a是shell任务，task_b是switch任务
+### 1、创建工作流
```
Comment (QuakeWang) [inline docs/docs/zh/guide/task/switch.md]:
```suggestion
### 1、创建工作流

```
It is recommended to add a line break between the title and body statement. The format will be more standardized. The same as following.

File: docs/docs/en/guide/task/switch.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # Switch
 
 The switch is a conditional judgment node, decide the branch executes according to the value of [global variable](../parameter/global.md) and the expression result written by the user.
+**Note** Execute expressions using javax.script.ScriptEngine.eval.
 
 ## Create
```
Comment (QuakeWang) [inline docs/docs/en/guide/task/switch.md]:
```suggestion
## Create Task
```

File: docs/docs/en/guide/task/switch.md [hunk 6]
```diff
@@ -14,26 +15,33 @@ Drag from the toolbar <img src="/img/switch.png" width="20"/>  task node to canv
...
+- Environment name: The environment in Security, if not configured, it will not be used.
+- Task group name: The group in Resources, if not configured, it will not be used.
 - Times of failed retry attempts: The number of times the task failed to resubmit. You can select from drop-down or fill-in a number.
 - Failed retry interval: The time interval for resubmitting the task after a failed task. You can select from drop-down or fill-in a number.
+- Delay execution time: Task delay execution time.
 - Timeout alarm: Check the timeout alarm and timeout failure. When the task runs exceed the "timeout", an alarm email will send and the task execution will fail.
 - Condition: You can configure multiple conditions for the switch task. When the conditions are satisfied, execute the configured branch. You can configure multiple different conditions to satisfy different businesses.
 - Branch flow: The default branch flow, when all the conditions are not satisfied, execute this branch flow.
 
 ## Detail
```
Comment (QuakeWang) [inline docs/docs/en/guide/task/switch.md]:
```suggestion
## Task Example
```

File: docs/docs/en/guide/task/switch.md (hunk 5 above)
Comment (QuakeWang) [reply docs/docs/en/guide/task/switch.md]: - Click `Project -> Management-Project -> Name-Workflow Definition`, and click the Create Workflow button to enter the DAG editing page.
- Drag from the toolbar  to the canvas.

File: docs/docs/en/guide/task/switch.md [hunk 7]
```diff
@@ -14,26 +15,33 @@ Drag from the toolbar <img src="/img/switch.png" width="20"/>  task node to canv
 - Descriptive information: Describe the function of the node.
```
Comment (QuakeWang) [inline docs/docs/en/guide/task/switch.md:16]: Remove extra indent spaces

**Review by QuakeWang — COMMENTED** [review]
The provided images can be marked with corresponding tags at key information. In this way, it can be more beneficial to users.

**Comment by huangchenguang123** [comment]
@QuakeWang Thanks.I had fixed those problems.

File: docs/docs/en/guide/task/switch.md [hunk 8]
```diff
@@ -14,26 +15,42 @@ Drag from the toolbar <img src="/img/switch.png" width="20"/>  task node to canv
...
 - Branch flow: The default branch flow, when all the conditions are not satisfied, execute this branch flow.
 
-## Detail
+## Task Example
+
+This is demonstrated using one switch task and three shell tasks.
+
+![switch_01](/img/tasks/demo/switch_01.png)
+
+### 1. Create a workflow
```
Comment (zhongjiajie) [inline docs/docs/en/guide/task/switch.md]: We should avoid using number in markdown title, the prefix `###` is for the same function @huangchenguang123  and cc @QuakeWang 
```suggestion
### Create a workflow
```

Comment (zhongjiajie) [reply docs/docs/en/guide/task/switch.md]: same as others places

File: docs/docs/zh/guide/task/switch.md [hunk 9]
```diff
@@ -1,37 +1,57 @@
...
+如果执行正确，那么 taskA 会被正确执行。
 
-最终switch任务的配置如下
+### 3、执行
 
-![task-switch-configure](/img/switch_configure.jpg)
+执行，并且查看是否符合预期。可以看到符合预期，执行了指定的下游任务 taskA。
 
-## 相关任务
+![switch_04](../../../../img/tasks/demo/switch_04.png)
```
Comment (zhongjiajie) [inline docs/docs/zh/guide/task/switch.md:57]: Maybe we should still using the old style start with `/img` until we merge https://github.com/apache/dolphinscheduler-website/pull/789 WDYT @huangchenguang123 
```suggestion
![switch_04](/img/tasks/demo/switch_04.png)
```

Comment (huangchenguang123) [reply docs/docs/zh/guide/task/switch.md:57]: This is my problem.I forgot to delete it after testing.

**Comment by huangchenguang123** [comment]
I had fixed those problems.

File: docs/docs/zh/guide/task/switch.md [hunk 10]
```diff
@@ -1,37 +1,57 @@
...
+- 环境名称：安全中心中配置的环境，不配置则不使用。
+- 任务组名称：资源中心中配置的任务组，不配置则不使用。
 - 失败重试次数：任务失败重新提交的次数，支持下拉和手填。
 - 失败重试间隔：任务失败重新提交任务的时间间隔，支持下拉和手填。
-- 超时告警：勾选超时告警、超时失败，当任务超过"超时时长"后，会发送告警邮件并且任务执行失败.
-- 条件：可以为switch任务配置多个条件，当条件满足时，就会执行指定的分支，可以配置多个不同的条件来满足不同的业务
-- 分支流转：默认的流转内容，当**条件**中的内容为全部不符合要求时，则运行**分支流转**中指定的分支
+- 延时执行时间：任务延迟执行的时间。
+- 超时告警：勾选超时告警、超时失败，当任务超过"超时时长"后，会发送告警邮件并且任务执行失败。
+- 条件：可以为switch任务配置多个条件，当条件满足时，就会执行指定的分支，可以配置多个不同的条件来满足不同的业务，使用字符串判断时需要使用""。
```
Comment (QuakeWang) [inline docs/docs/zh/guide/task/switch.md]:
```suggestion
- 条件：可以为 switch 任务配置多个条件，当条件满足时，就会执行指定的分支，可以配置多个不同的条件来满足不同的业务，使用字符串判断时需要使用""。
```
Add a space between Chinese and English.

**Comment by zhongjiajie** [comment]
I find out we still have a discussion in https://github.com/apache/dolphinscheduler/pull/10247#discussion_r882268874 and I do not have much experience for switch tasks. Maybe you guys should discuss on it

**Comment by huangchenguang123** [comment]
> I find out we still have a discussion in [#10247 (comment)](https://github.com/apache/dolphinscheduler/pull/10247#discussion_r882268874) and I do not have much experience for switch tasks. Maybe you guys should discuss on it

What do you think about it?Do you have more suggestions? @QuakeWang

**Review by QuakeWang — DISMISSED** [review]
This PR sets the corresponding switch branch using global variables. In this way, users can intuitively understand how to use this task.

**Review by zhongjiajie — DISMISSED** [review]
Thanks @huangchenguang123 and @QuakeWang for the review

**Comment by zhongjiajie** [comment]
I solve some upstream change in branch dev

File: docs/docs/en/guide/task/switch.md [hunk 11]
```diff
@@ -14,26 +16,42 @@ Drag from the toolbar <img src="../../../../img/switch.png" width="20"/>  task n
...
 
-The following shows the switch task configuration:
+### Create a workflow
 
-![task-switch-configure](../../../../img/switch_configure.jpg)
+Create a new switch task, and three shell tasks downstream. The shell task is not required.
+The switch task needs to be connected with the downstream task to configure the relationship before the downstream task can be selected.
 
-## Related Task
+![switch_01](/img/tasks/demo/switch_01.png)
```
Comment (zhongjiajie) [inline docs/docs/en/guide/task/switch.md]: You should use related path instead of absolute path after #10325 merged

**Review by zhongjiajie — CHANGES_REQUESTED** [review]
img path usage

File: docs/docs/en/guide/task/switch.md (hunk 11 above)
Comment (huangchenguang123) [reply docs/docs/en/guide/task/switch.md]: I had change the img path.

**Comment by huangchenguang123** [comment]
PTAL @zhongjiajie

**Review by zhongjiajie — COMMENTED** [review]
Looking great, thanks

**Review by zhongjiajie — APPROVED** [review]

**Pull request merged**